thoth-api/src/model/publication/crud.rs

```diff
@@ -169,14 +169,27 @@ impl DbInsert for NewPublicationHistory {
     db_insert!(publication_history::table);
 }
 
-impl NewPublication {
-    pub fn can_have_isbn(&self, db: &crate::db::PgPool) -> ThothResult<()> {
+pub trait PublicationValidation
+where
+    Self: PublicationProperties,
+{
+    fn can_have_isbn(&self, db: &crate::db::PgPool) -> ThothResult<()>;
```

Comment:
this method could also have a default implementation. If the problem is that not all PublicationProperties implementers will have a work_id we can still have a work_id() method marked as `umimplemented!()` in those cases